File: src/ios/ee/ads/MessageHelper.swift
//
//  MessageHelper.swift
//  ee-x-366f64ba
//
//  Created by eps on 6/24/20.
//

import Foundation

class MessageHelper {
    private let _prefix: String
    private let _adId: String

    init(_ prefix: String, _ adId: String) {
        _prefix = prefix
        _adId = adId
    }

    var isLoaded: String { return "\(_prefix)_isLoaded_\(_adId)" }
    var load: String { return "\(_prefix)_load_\(_adId)" }
    var onLoaded: String { return "\(_prefix)_onLoaded_\(_adId)" }
    var onFailedToLoad: String { return "\(_prefix)_onFailedToLoad_\(_adId)" }
    var show: String { return "\(_prefix)_show_\(_adId)" }
    var onFailedToShow: String { return "\(_prefix)_onFailedToShow_\(_adId)" }
    var onClicked: String { return "\(_prefix)_onClicked_\(_adId)" }
    var onClosed: String { return "\(_prefix)_onClosed_\(_adId)" }
    var getPosition: String { return "\(_prefix)_getPosition_\(_adId)" }
    var setPosition: String { return "\(_prefix)_setPosition_\(_adId)" }
    var getSize: String { return "\(_prefix)_getSize_\(_adId)" }
    var setSize: String { return "\(_prefix)_setSize_\(_adId)" }
    var isVisible: String { return "\(_prefix)_isVisible_\(_adId)" }
    var setVisible: String { return "\(_prefix)_setVisible_\(_adId)" }
    var createInternalAd: String { return "\(_prefix)_createInternalAd_\(_adId)" }
    var destroyInternalAd: String { return "\(_prefix)_destroyInternalAd_\(_adId)" }
}

File: src/ts/src/ads/internal/MessageHelper.ts
export class MessageHelper {
    private readonly _prefix: string;
    private readonly _adId: string;

    public constructor(prefix: string, adId: string) {
        this._prefix = prefix;
        this._adId = adId;
    }

    public get isLoaded(): string {
        return `${this._prefix}_isLoaded_${this._adId}`;
    }

    public get load(): string {
        return `${this._prefix}_load_${this._adId}`;
    }

    public get onLoaded(): string {
        return `${this._prefix}_onLoaded_${this._adId}`;
    }

    public get onFailedToLoad(): string {
        return `${this._prefix}_onFailedToLoad_${this._adId}`;
    }

    public get show(): string {
        return `${this._prefix}_show_${this._adId}`;
    }

    public get onFailedToShow(): string {
        return `${this._prefix}_onFailedToShow_${this._adId}`;
    }

    public get onClicked(): string {
        return `${this._prefix}_onClicked_${this._adId}`;
    }

    public get onClosed(): string {
        return `${this._prefix}_onClosed_${this._adId}`;
    }

    public get getPosition(): string {
        return `${this._prefix}_getPosition_${this._adId}`;
    }

    public get setPosition(): string {
        return `${this._prefix}_setPosition_${this._adId}`;
    }

    public get getSize(): string {
        return `${this._prefix}_getSize_${this._adId}`;
    }

    public get setSize(): string {
        return `${this._prefix}_setSize_${this._adId}`;
    }

    public get isVisible(): string {
        return `${this._prefix}_isVisible_${this._adId}`;
    }

    public get setVisible(): string {
        return `${this._prefix}_setVisible_${this._adId}`;
    }
}